core/pta/system.c

```diff
@@ -856,14 +855,13 @@ static TEE_Result system_supp_plugin_invoke(uint32_t param_types,
 	if (exp_pt != param_types)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	tee_uuid_from_octets(&op.uuid, (const uint8_t *)params[0].memref.buffer);
-
-	op.cmd = params[1].value.a;
-	op.sub_cmd = params[1].value.b;
-	op.data = params[2].memref.buffer;
-	op.len = params[2].memref.size;
-
-	return tee_invoke_supp_plugin_rpc(&op);
+	return tee_invoke_supp_plugin_rpc(params[0].memref.buffer, /* uuid */
+					  params[1].value.a, /* cmd */
+					  params[1].value.b, /* sub_cmd */
+					  params[2].memref.buffer, /* data */
+					  params[2].memref.size, /* in len */
+					  (size_t *)&params[3].value.a
```

Comment:
This cast isn't portable.

Reply:
Yes, you are right. I will fix it

Reply:
I'm sorry, I should have been more clear. Please use a temporary helper variable instead of changing the type in the argument of `tee_invoke_supp_plugin_rpc()`.

Reply:
It seems you see outdated changes.
In previous commits I changed `outlen` type from `size_t *` to `uint32_t *` and removed type conversion in  `core/pta/system.c:863`. See the last fixup commit: https://github.com/OP-TEE/optee_os/pull/4248/commits/3a46040dbf438d47a4301c1d3c5bae53446b9315#diff-80ee4f614b595d036997a8c22e8034f37acf85250c5829593e630cab71a9bb4aR863.

I can return back `size_t *` and add temporary helper variable, if it will be better.

Reply:
> I can return back size_t * and add temporary helper variable, if it will be better.

Please do so.